File: c++/1.cpp
#include <iostream>
#include <string>

int strStr(std::string haystack, std::string needle) {
   for (int i = 0; i <= haystack.length() - needle.length(); i++) {
       if (haystack.substr(i, needle.length()) == needle) {
           return i;
       }
   }
   return -1;
}

int main() {
   std::string haystack[] = {"bassadbas", "sleetcodes", "github"};
   std::string needle[] = {"sad", "leetcode", "ggithub"};

   std::cout << strStr(haystack[0], needle[0]) << std::endl;
   std::cout << strStr(haystack[1], needle[1]) << std::endl;
   std::cout << strStr(haystack[2], needle[2]) << std::endl;

   return 0;
}

File: rust/1.rs
fn str_str(haystack: &str, needle: &str) -> i32 {
   for i in 0..=haystack.len() - needle.len() {
       if &haystack[i..i+needle.len()] == needle {
           return i as i32;
       }
   }
   return -1;
}

fn main() {
   let haystack = ["bassadbas", "sleetcodes", "github"];
   let needle = ["sad", "leetcode", "ggithub"];

   println!("{}", str_str(haystack[0], needle[0]));
   println!("{}", str_str(haystack[1], needle[1]));
   println!("{}", str_str(haystack[2], needle[2]));
}